Ark.Tools.Sql/AbstractSqlContextAsync.cs

```diff
@@ -0,0 +1,110 @@
+            _transaction = transaction;
+        }
+
+        public AbstractSqlContextAsync(DbConnection connection, IsolationLevel isolationLevel = IsolationLevel.ReadCommitted)
+        {
+            _connection = connection ?? throw new ArgumentNullException(nameof(connection));
+            _isolationLevel = isolationLevel;
+            _transaction = _connection.BeginTransaction(isolationLevel);
+        }
+
+        public AbstractSqlContextAsync(DbTransaction transaction)
+        {
+            _transaction = transaction ?? throw new ArgumentNullException(nameof(transaction));
+            _connection = transaction.Connection ?? throw new ArgumentNullException(nameof(transaction.Connection));
+            _isolationLevel = transaction.IsolationLevel;
+        }
+
+
+        public DbConnection Connection
+        {
+            get
+            {
+                return _connection ?? throw new InvalidProgramException("There's no connection");
+            }
+        }
+
+        public DbTransaction Transaction
+        {
+            get
+            {
+                return _transaction ?? throw new InvalidProgramException("There's no transaction");
+            }
+        }
+
+        public virtual async ValueTask CommitAysnc(CancellationToken ctk)
+        {
+#if !(NETSTANDARD2_0 || NET472)
+            if (_transaction != null)
+            {
+                await _transaction.CommitAsync(ctk);
+                await _transaction.DisposeAsync();
+            }
+            _transaction = null;
+#else
+            throw new NotSupportedException("Async SQL not supported for this .NET framework");
+#endif
+        }
+
+        public async ValueTask ChangeIsolationLevelAsync(IsolationLevel isolationLevel, CancellationToken ctk)
+        {
+#if !(NETSTANDARD2_0 || NET472)
+            _isolationLevel = isolationLevel;
+            await RollbackAsync(ctk);
+#else
+            throw new NotSupportedException("Async SQL not supported for this .NET framework");
+#endif
+        }
+
+        public async ValueTask RollbackAsync(CancellationToken ctk)
+        {
+#if !(NETSTANDARD2_0 || NET472)
+            if (_transaction != null)
+            {
+                await _transaction.RollbackAsync(ctk);
+                await _transaction.DisposeAsync();
+            }
+
+            _transaction = null;
+#else
+            throw new NotSupportedException("Async SQL not supported for this .NET framework");
+#endif
+        }
+
+        public async ValueTask DisposeAsync()
+        {
+#if !(NETSTANDARD2_0 || NET472)
+            if (_transaction != null)
+            {
+                await _transaction.DisposeAsync();
+            }
+
+            if (_connection != null)
+            {
+                await _connection.DisposeAsync();
+            }
+            GC.SuppressFinalize(this);
+#else
+            throw new NotSupportedException("Async SQL not supported for this .NET framework");
```

Comment:
as already said, this pattern is not OK.

it's not that each 'method' should throw on old frameworks, but the _whole_ Async class should not exists. the whole file must be empty in that case.

on a separate note, starting v5, I've deprecated some Frameworks. please check DirectoryBuild to see if this is still required (it was on master)